src/strategies/base.ts

```diff
@@ -254,30 +268,145 @@ export abstract class BaseStrategy implements Strategy {
+    latestRelease,
+    draft,
+    manifestPath,
+  }: {
+    commits: ConventionalCommit[];
+    latestRelease?: Release;
+    draft?: boolean;
+    labels?: string[];
+    existingPullRequest?: PullRequest;
+    manifestPath?: string;
+  }): Promise<ReleasePullRequest | undefined> {
     const conventionalCommits = await this.postProcessCommits(commits);
     this.logger.info(`Considering: ${conventionalCommits.length} commits`);
     if (conventionalCommits.length === 0) {
       this.logger.info(`No commits for path: ${this.path}, skipping`);
       return undefined;
     }
 
-    const newVersion = await this.buildNewVersion(
-      conventionalCommits,
-      latestRelease
+    const component = await this.getComponent();
+    this.logger.debug('component:', component);
+
+    const releaseAsCommit = conventionalCommits.find(conventionalCommit =>
+      conventionalCommit.notes.find(note => note.title === 'RELEASE AS')
+    );
+    const releaseAsNote = releaseAsCommit?.notes.find(
+      note => note.title === 'RELEASE AS'
     );
+
+    let newVersion: Version;
+    if (this.releaseAs) {
+      this.logger.warn(
+        `Setting version for ${this.path} from release-as configuration`
+      );
+      newVersion = Version.parse(this.releaseAs);
+    } else if (releaseAsNote) {
+      newVersion = Version.parse(releaseAsNote.text);
+    } else if (latestRelease) {
+      newVersion = await this.versioningStrategy.bump(
+        latestRelease.tag.version,
+        conventionalCommits
+      );
+    } else {
+      newVersion = this.initialReleaseVersion();
+    }
+
+    // If a pull request already exists, compare the manifest version from its branch against the one from the PR title.
+    // If they don't match, assume the PR title has been edited by an end user to set the version.
+    if (existingPullRequest) {
+      this.logger.info(
+        `PR already exists for ${existingPullRequest.headBranchName}, checking if PR title edited to set custom version`
+      );
+      const existingPRTitleVersion = PullRequestTitle.parse(
+        existingPullRequest.title
+      )?.getVersion();
+      const hasCustomVersionLabel = existingPullRequest.labels.find(
+        label => label === DEFAULT_CUSTOM_VERSION_LABEL
+      );
+
+      if (!existingPRTitleVersion && hasCustomVersionLabel) {
+        // report problem to end user
+        this.github.commentOnIssue(
+          `
+## Invalid version number in PR title
+
+:rotating_light: This Pull Request has the \`${DEFAULT_CUSTOM_VERSION_LABEL}\` label but the version number cannot be found in the title. Instead the generated version \`${newVersion}\` will be used.
+
+If you want to use a custom version be sure to use the semver format.
+
+If you do not want to set a custom version and want  to get rid of this warning, remove the label \`${DEFAULT_CUSTOM_VERSION_LABEL}\` from this Pull Request.
+`,
+          existingPullRequest.number
+        );
+      } else if (!existingPRTitleVersion) {
```

Comment:
Can we differentiate between no version or an invalid version? (if not, don't bother, not important enough)

Reply:
Hmm, not without some changes to the parsing.

It's currently parsed in two steps:
1. a match with `/v?(?<version>[0-9].*)/`
2. followed by matching with `/(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)(-(?<preRelease>[^+]+))?(\+(?<build>.*))?/`

We could potentially report "invalid format" if the first match is successful but not the second one, and "missing version" if the first match fails? Not sure if it is worth changing the parsing for this but is it feasible.